plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -34,4 +34,105 @@ import NotificationCenter
         guard let config = OptimoveSDKPlugin.config else { return }
         Optimove.initialize(with: config)
     }
+    
+    @objc(reportEvent:)
+    func reportEvent(command: CDVInvokedUrlCommand) {
+        var name = ""
+        var parameters: [String: Any] = [:]
+        
+        if let n = command.arguments[0] as? String {
+            name = n
+        }
+        
+        if let param = command.arguments[1] as? [String: Any] {
+            parameters = param
+        }
+        
+        Optimove.shared.reportEvent(name: name, parameters: parameters)
+        
+        self.commandDelegate.run {
```

Comment:
In all of these methods I'm not sure if we need to run this on the delegate or not. The basic plugin example doesn't post the callbacks back to the delegate queue.

Is there a particular reason to post these back, or do you think we can safely remove the `self.commandDelegate.run` calls?